### PR TITLE
Migrate to kubebuilder

### LINKED ISSUE #252: Generate CRDs with kubebuilder
/kind feature

**Describe the solution you'd like**
Instead of copy/pasting CRDs from the gcp project we should generate them ourselves

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Cluster-api-provider-aws version: 
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`): @

### PR DESCRIPTION
**Pull request by chuckha**
* Spec/Status is now housed under one ProviderConfig
* Controllers and types generated with kubebuilder
* CRDs generated with kubebuilder

**What this PR does / why we need it**:
This PR migrates our code to be in line with kubebuilder expectations. This has a few side effects such as reducing some boiler plate code.

The controllers are responsible for updating the objects so we no longer have to manage this ourselves.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #252 

**Special notes for your reviewer**:
I'm happy to walk through this with you if you like

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/apis/awsproviderconfig/v1alpha1/awsclusterproviderconfig_types.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+type AWSClusterProviderConfigSpec struct {
+	// The AWS Region the cluster lives in.
+	Region string `json:"region"`
+
+	// SSHKeyName is the name of the ssh key to attach to the bastion host.
+	SSHKeyName string `json:"sshKeyName,omitempty"`
+}
+
+// AWSClusterProviderConfigStatus defines the observed state of AWSClusterProviderConfig
+type AWSClusterProviderConfigStatus struct {
```
Comment (detiber) [inline pkg/apis/awsproviderconfig/v1alpha1/awsclusterproviderconfig_types.go]: I'm not sure we should overload the providerconfig to include a persisted status here...

Comment (chuckha) [reply pkg/apis/awsproviderconfig/v1alpha1/awsclusterproviderconfig_types.go]: oh hmm because the Cluster would have a status already?

**Comment by chuckha** [comment]
/hold

so i can get some feedback

**Comment by chuckha** [comment]
oops, close but this is not what we want. close though!

**Comment by chuckha** [comment]
oops, close but this is not what we want. close though!

**Comment by vincepri** [comment]
@chuckha can you elaborate a little please?

File: pkg/apis/awsproviderconfig/v1alpha1/awsclusterproviderconfig_types.go (hunk 1 above)
Comment (detiber) [reply pkg/apis/awsproviderconfig/v1alpha1/awsclusterproviderconfig_types.go]: We'd be conflating spec/status concepts and would be splitting our status between the actual Cluster/Machine Status and Spec.ProviderConfig

**Pull request closed without merging**